### ISSUE TITLE
Perspective breaks after enable-theme

### ISSUE THREAD

**Issue by Thuna-Cing** [nex3/perspective-el#185]
Whenever `load-theme` or `enable-theme` is called, all perspectives get discarded and a "main" perspective is (re)created as though `persp-mode` was restarted.  The issue persists with `emacs -Q`

Emacs version 29.0.50 & Perspective commit 794afdbc5188ef6f2d78d26302cd78903ce618fa

**Comment by gcv** [comment]
Everything works for me with Emacs 28.1. My system currently has trouble compiling Emacs from the master branch, and I don’t have time to deal with it, so I can’t work on reproducing the problem right now.

Please check if 28.1 works for you.

**Comment by Thuna-Cing** [comment]
I can confirm that the issue is still present with Emacs 28.1 on a different device.

**Comment by gcv** [comment]
Something strange is going on. I'm using the following invocation to start Emacs (28.1, tried with and without native compilation turned on):

```shell
/path/to/emacs -nw -Q -l /path/elpa/perspective-20220521.2138/perspective-autoloads.el
```

Then I paste the following code into the `*scratch*` buffer and run it:

```elisp
(progn
  (customize-set-variable 'persp-mode-prefix-key (kbd "C-c C-p"))
  (persp-mode 1)
  (persp-switch "five")
  (with-current-buffer "*scratch* (five)"
    (insert "in perspective five"))
  (persp-switch "main")
  (load-theme 'wombat nil)
  )
```

This works as expected. The `five` perspective is _not_ discarded as you describe.

1. What happens when you run this code with the given start command?
2. If the problem occurs for you, take a look at your Perspective installation. Could there be stale .elc files floating around?
3. Try to (1) fully uninstall Perspective, (2) restart Emacs, and (3) install Perspective again.

Please let me know what you find. I'd like to get to the bottom of this problem.

**Comment by Thuna-Cing** [comment]
After some testing, I have tracked down the issue to a specific
decision: whether persp-mode is called interactively or not.  If
persp-mode is activated via ```(persp-mode 1)```, the issue is not observed.
If it is activated via ```M-x persp-mode``` or ```(call-interactively 'persp-mode)```,
it is.  Any subsequent attempts with ```(persp-mode 1)``` will also fail.

 The final testing was done with the following code:

 ```elisp
 (load "/path/to/perspective/perspective-autoloads.el" nil nil t)
 (add-to-list 'load-path "/path/to/perspective/")

 (persp-mode 0)

 (persp-mode 1)
 (call-interactively 'persp-mode)

 (progn
   (persp-switch "five")
   (with-current-buffer "*scratch* (five)"
     (insert "in perspective five"))
   (persp-switch "main")
   (load-theme 'wombat nil)
   (disable-theme 'wombat))
 ```

 by executing the forms in order (I did it with ```C-x C-e```):

 ```elisp
 (persp-mode 1)
 (progn...)
 ;; Here, the perspective "five" will be present

 (persp-mode 0)
 (persp-mode 1)
 (progn...)
 ;; Here, the perspective "five" will still be present
 ;; This can be repeated as many times as needed without effecting the
 ;; outcome.

 (persp-mode 0)
 (call-interactively 'persp-mode) ; or [M-x persp-mode]
 (progn...)
 ;; Here, the perspective "five" will not be present

 (persp-mode 0)
 (persp-mode 1)
 (progn...)
 ;; Here, the perspective "five" will still not be present
 ;; This can be repeated as many times as needed without effecting the
 ;; outcome.

**Comment by gcv** [comment]
Excellent. That helped me reproduce the problem and track it down. It should be fixed in commit b6a13bf, which will hit MELPA in a couple of hours. Please let me know if that resolves the bug for you.

In case you're curious, it happened because repeatedly activating `persp-mode` causes it to reinitialize itself. This would occur if you called `(persp-mode 1)` twice in a row. No one does that in practice, which is why the problem never came up before. However, if you activate `persp-mode` interactively, it gets registered with the "custom" framework (I don't fully understand how). When `load-theme` gets called, its integration with "custom" causes it to reinitialize `persp-mode`, effectively calling `(persp-mode 1)` a second time and triggering reinitialization.

The (attempted) fix is to make activating `persp-mode` an idempotent operation.

Thank you for taking the time to find a reproducible test case I could work with.
